### DISCUSSION TITLE
Continue to custom.domain instead of XXXX.supabase.co when using Google as provider

### DISCUSSION THREAD

**Discussion by AdrianKrebs** [supabase/supabase#5041] [Questions] [answered]
When users log in with Google, they see the modal "Continue to "asdasdasdad.supabase.co", which looks not trustworthy for users.
How can I change this?

I found two related issues, talking about URL rewrites, but by default, supabase redirects to "https://asdasdasdad.supabase.co/auth/v1/authorize".

How do I change that?

Related:
https://github.com/supabase/supabase/issues/1798#issuecomment-853546270
https://github.com/supabase/gotrue/issues/142

**Comment by kangmingtay** [comment]
Hi @AdrianKrebs, there isn't a way to use custom domains for oauth logins for now, but we'll be looking to add custom domains soon (possibly Q2 or Q3 this year)!

**Reply by AdrianKrebs** [reply]
Thanks, @kangmingtay. I saw quite some confusion and discussion around this. Many developers complain that it just looks untrustworthy/scammy to get redirected to an unknown domain. Maybe a short sentence in the Supabase Google  OAuth docs could clarify this?

**Reply by silentworks** [reply]
@AdrianKrebs for now you can do what is suggested in the thread linked below for Google to use your app name when you decide to go into production. https://github.com/supabase/supabase/discussions/2925#discussioncomment-1716295

**Reply by AdrianKrebs** [reply]
@silentworks  thanks, I'll try it, even though it sounds like a pretty frustrating process with Google: https://github.com/supabase/supabase/discussions/2925#discussioncomment-1999792

**Reply by mirkonasato** [reply]
Just attempted the Google OAuth Verification process, and failed. It's no longer possible to simply remove the `*.supabase.co` domain from the list as suggested in the thread linked by @silentworks above.

The situation right now is that you either show your users a prompt saying something like "continue to brgfggfwfwbbswmdhrgr.supabase.co", or don't use Google sign in at all.

**Comment by activenode** [comment]
A client of mine says that their spam detector doesn't like the supabase.co stuff and that it's being problematic. I need this feature too.

**Reply by madeleineostoja** [reply]
There’s a waitlist for custom domains as an addon, email supabase

**Comment by pedromassango** [comment]
Did we get any update on this? It is pretty frustrating to use the Google process. Custom domain for auth should be part of supabase  paying customers.

**Comment by kangmingtay** [comment, marked as answer]
Hey everyone, this is now possible by adding a custom domain to your Supabase project: https://supabase.com/docs/guides/platform/custom-domains

You can also verify your app to display a custom name on the OAuth screen if you don't want to pay for the custom domain.
https://support.google.com/cloud/answer/13463073?hl=en

**Reply by pedromassango** [reply]
I did tried this, however, Google was redirecting to my website instead.

For context, my domain is `example.app`. When I add the subdomain `api.example.app` in supabase custom domain, Google redirects to my website  instead of using the supabase callback.

**Reply by kangmingtay** [reply]
did you remember to add your custom domain as the callback URL in your google oauth app though?

**Reply by pedromassango** [reply]
Yes

**Reply by kangmingtay** [reply]
@pedromassango reach out to us via support at https://app.supabase.com/support/new and we'll investigate the issue with your project!

**Reply by helloromero** [reply]
$10/month is a bit much for me 😢

I guess I will try to verify via Google and see how it goes...  Google says "The review process can take up to 4–6 weeks" so lets see...

**Reply by KPR111** [reply]
> $10/month is a bit much for me 😢
> 
> I guess I will try to verify via Google and see how it goes... Google says "The review process can take up to 4–6 weeks" so lets see...

hey.. I tried yousing google ... and it's even verified .. but even though in google sign in it's still showing like supabase.co...
Do you have any solution
